### ISSUE TITLE
[UX] Move title display settings to Manage Display form

### ISSUE THREAD

**Issue by jenlampton** [backdrop/backdrop-issues#1950]
It is non-intuitive for people to adjust the way the title behaves on the "Configure" page for content types when all the other fields are managed via `Manage fields` and `Manage display`. We should move these settings over to where people expect to find them, using `hook_field_extra_fields()` in the node module.

1) Everything in the `Submission form settings` tab should be moved to `Manage fields`
- The `Title field label` options should be handled the same way as other field titles (edit button, with inner label field)
- The `Explanation or submission guidelines` should also be moved, perhaps to the top of the list

2) A new option should be added to the `Manage display` tab to show or hide the title.

(also, I thought we had an issue for this already, but perhaps I am thinking of the Drupal issue I created years ago: https://www.drupal.org/node/1513054)

**Comment by klonos** [comment]
For reference, the respective issue in d.org against D8: [D8: Title does not appear as a field in Manage Display [#2353867]](https://www.drupal.org/node/2353867)

**Comment by klonos** [comment]
> 2) A new option should be added to the `Manage display` tab to show or hide the title.

We shouldn't special-case the title. There is already such an option for the rest of the fields in the "Format" drop-down menu that can be used for that purpose:

![backdrop-content_type_manage_display_tab-hidden_format_option](https://cloud.githubusercontent.com/assets/2423362/15742296/5324ed7e-2901-11e6-8a39-576034a58204.png)

...plus there's the alternative of simply dragging the field down to the "Hidden" area.

**Comment by klonos** [comment]
The summary from the d.org issue:

> When I go to Content Types, and select the Manage Fields tab, I was surprise to **not** see my node Title field, so that I can chose its location, whether to display its label, and whether it is displayed or hidden.

I bet we'll have the same reaction when people will not be able to find the title in the "Manage fields" tab (:cough: #933 :cough:).

**Comment by klonos** [comment]
> (also, I thought we had an issue for this already, but perhaps I am thinking of the Drupal issue I created years ago: https://www.drupal.org/node/1513054)

#1441 ???

**Comment by klonos** [comment]
...oh, you propose to put this in the Manage display tab. My proposal in #1441 is to put it in the "Manage fields" tab. So lets keep this open then and decide which way to go.

**Comment by Bojhan** [comment]
@klonos Your suggestion might not be very pragmatic. The title is not yet a field. If you want it to leverage the "manage fields display" screen you will need to fake it into being a field. 

I think my won't fix many years ago, was in expectation it would be nicely a field. That never happens so this approach makes sense. 

Please don't close issues without clear reasoning.

**Comment by klonos** [comment]
>Please don't close issues without clear reasoning.

@Bojhan I closed this issue because I *thought* it was a duplicate of #1441. That was my clear reasoning. I reopened it when I realized that the two issues are actually different proposals.

**Comment by sthomen** [comment]
Is anything happening here? Is there a particular sticking point?

I regularly use display suite (which is rather overkill) with drupal7 just to hide or move the title field when nodes are rendered in views. I can't do that in backdrop without making custom node template files. You've implemented half of the features I use from display suite (custom display types), I'd love to have the other half (the ability to move/hide everything in a node without custom templates).

I could probably do it myself, but I am not terribly familiar with backdrop yet.

**Comment by jenlampton** [comment]
Hi @sthomen,

The problem with the node title is that it's already treated specially in the node template in Backdrop (and Drupal) core. Any changes we make via Manage displays would be very likely to break all the current sites, since all the templates already handle the title specially.

This issue is to move the settings for the title field to the manage displays form, but will not change the rendering of the title. It sounds like you may be asking about the rendering, as well.

It would be possible to provide a contrib module that allows for special rendering of the node title (to make it more like other fields!) but it will be very hard to make that kind of change to core without adversely affecting everyone else. I suspect a change that drastic will need to wait for Backdrop 2.x.

Would you please add a request for the title field handling from display suite over in https://github.com/backdrop-ops/contrib/issues?  I expect you are not the only one who wants that feature.

**Comment by olafgrabienski** [comment]
@sthomen, in the following issue you'll find more information regarding the title display:
- #933 - [UX] Make node title a proper (sortable too) field. a.k.a. Title module in core

Also related:
- #1944 - Title Module Alternative Strategies (closed)
- #2084 - [META] [UX] Add more Display Suite features

**Comment by klonos** [comment]
@sthomen not sure if that module was solving what you are after, but [Title](https://www.drupal.org/project/title) has already been ported: https://github.com/backdrop-contrib/title 😉

**Comment by jenlampton** [comment]
Hi @sthomen,
there's no need to appologize. I misunderstood this issue too, until I re-read all the comments. It happens all the time! We appreciate your participation :)

**Comment by BWPanda** [comment]
I think this may be blocked by https://github.com/backdrop/backdrop-issues/issues/3859 since pseudo (extra) fields can't have settings...

**Comment by jenlampton** [comment]
Also, I'm pretty sure this issue is a dupe of https://github.com/backdrop/backdrop-issues/issues/1441. Closing this in favor of the earlier one.

**Comment by jenlampton** [comment]
Oh hold on... this is `Manage Display form` and the other is `Manage Fields` form... reopening :)

**Comment by BWPanda** [comment]
Ha, @klonos thought the same thing 3 years ago 😄

**Comment by quicksketch** [comment]
I wrote a contrib module to help with this need while we figure this out in core: https://github.com/backdrop-contrib/title_display

I am still 100% in favor of this being a core-provided feature. I cross linked this issue from the README.md.

**Comment by muranod** [comment]
I was just exploring that now, coming from Drupal 7/8/9 where I was using Display Suite to move the title, among other things.  The title_display module allows me to move the title, but not within the actual content fields. Mine is primarily a photo site and I want to display it (as in my current D8/9 site) with the content Title _under_ the Lede photo. I just think it looks more elegant that way. 
![sample-layout-of-my-webpage](https://github.com/backdrop/backdrop-issues/assets/15290888/f2fe3045-32b7-4d3f-832b-2f54615870a0)

**Comment by quicksketch** [comment]
@muranod [Title Display](https://github.com/backdrop-contrib/title_display) should allow you to do exactly that! Though I see that there is this issue: https://github.com/backdrop-contrib/title_display/issues/8, where the option doesn't show up on the default display mode. Maybe see if that's the problem you're facing. Either way, post an issue to the Title Display issue queue if you can't figure it out.

**Comment by muranod** [comment]
Will do, thanks! (I thought I was posting in the Backdrop forum - sorry if I posted that in the wrong place.) - Dan

**Comment by stpaultim** [comment]
Is this issue still blocked by https://github.com/backdrop/backdrop-issues/issues/3859
